jsonschema-hyperschema.xml

```diff
@@ -1539,6 +1389,212 @@ GET /foo/
+                    <t>
+                        "ldo.templatePointers" is an empty object if the keyword was not
+                        present and "ldo.templateRequired" is likewise an empty array.
+                    </t>
+                    <t>
+                        "attachmentPointer" is the absolute JSON Pointer for the attachment
+                        location of the LDO.
+                    </t>
+                    <t>
+                        "getApplicableSchemas()" returns an iterable set of all (sub)schemas
+                        that apply to the attachment point in the instance.
+                    </t>
+                </list>
+            </t>
+            <t>
+                This algorithm should be applied first to either "href" or "anchor",
+                and then as needed to each successive "base".  The order is important,
+                as it is not always possible to tell whether a template will resolve
+                to a full URI or a URI-reference.
+            </t>
+            <t>
+                In English, the high-level algorithm is:
+                <list>
+                    <t>Populate template variable data from the instance</t>
+                    <t>If input is desired, accept input</t>
+                    <t>Check that all required variables have a value</t>
+                    <t>Encode values into strings and fill out the template</t>
+                </list>
+            </t>
+            <figure>
+                <preamble>
+                    This is the high-level algorithm as pseudocode.  "T" comes from either
+                    "href" or "anchor" within the LDO, or from "base" in a containing schema.
+                    Pseudocode for each step follows.
+                </preamble>
+                <artwork>
+<![CDATA[
+templateData = populateDataFromInstance(T, ldo, instance)
+
+if resolving "href" or "base" for "href" and ldo.hrefSchema exists:
+    inputData = acceptInput(ldo, instance, templateData)
+    for varname in inputData:
+        templateData[varname] = inputData[varname]
+
+for varname in ldo.templateRequired:
+    if not exists templateData[varname]
+        fatal("Missing required variable(s)")
+
+templateData = stringEncode(templateData)
+R = rfc6570ResolutionAlgorithm(T, templateData)
+]]>
+                </artwork>
+            </figure>
+
+            <section title="Populating template data from the instance">
+                <t>
+                    This step looks at various data sources in the instance:
+                    <list>
+                        <t>Use "templatePointers" to find values where appropriate</t>
+                        <t>Look in the immediate instance for remaining values</t>
+                        <t>
+                            Either way, if there is a value there, put it in the
+                            template resolution data set
+                        </t>
+                    </list>
+                </t>
```

Comment:
This paragraph (the whole `<t>`) does not look correct to me, or I'm not understanding it. It seems to me that it does not match with the pseudocode below. Specifically, I don't get the "Look in the immediate instance for remaining values".
Also I wouldn't say that "various data sources in the instance" are used (or, again, I may be missing what you mean by "various data sources"); to me the instance is just one source.